### PR TITLE
Add pop to bytearray

### PR DESCRIPTION
**Pull request by janczer**

### CODE REVIEW

File: vm/src/obj/objbytearray.rs [hunk 1]
```diff
@@ -261,3 +263,14 @@ fn bytearray_repr(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
...
 }
+
+fn bytearray_pop(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(obj, Some(vm.ctx.bytearray_type()))]);
+    let mut value = get_mut_value(obj);
+
+    if let Some(i) = value.pop() {
+        Ok(vm.ctx.new_int(i))
+    } else {
+        Err(vm.new_index_error("pop from empty bytearray".to_string()))
```
Comment (OddCoincidence) [inline vm/src/obj/objbytearray.rs]: Could you add a test to the python snippet that triggers this?

File: vm/src/obj/objbytes.rs [hunk 2]
```diff
@@ -188,6 +190,16 @@ pub fn get_value<'a>(obj: &'a PyObjectRef) -> impl Deref<Target = Vec<u8>> + 'a
     })
 }
 
+pub fn get_mut_value<'a>(obj: &'a PyObjectRef) -> impl DerefMut<Target = Vec<u8>> + 'a {
+    RefMut::map(obj.borrow_mut(), |py_obj| {
+        if let PyObjectPayload::Bytes { ref mut value } = py_obj.payload {
+            value
+        } else {
+            panic!("Inner error getting int {:?}", obj);
```
Comment (OddCoincidence) [inline vm/src/obj/objbytes.rs]:
```suggestion
            panic!("Inner error getting bytearray {:?}", obj);
```

Comment (janczer) [reply vm/src/obj/objbytes.rs]: Done

File: vm/src/obj/objbytearray.rs (hunk 1 above)
Comment (janczer) [reply vm/src/obj/objbytearray.rs]: Done

**Pull request merged**